blocks/identity-block/features/account-management/_children/PasswordEditableFieldContainer.jsx

```diff
@@ -0,0 +1,171 @@
+import passwordValidationMessage from '../../../utils/password-validation-message';
+import validatePasswordPattern from '../../../utils/validate-password-pattern';
+
+function PasswordEditableFieldContainer({ email, hasPassword }) {
+  const [error, setError] = useState(false);
+  const [passwordRequirements, setPasswordRequirements] = useState({
+    status: 'initial',
+  });
+
+  const { arcSite } = useFusionContext();
+  const { locale } = getProperties(arcSite);
+  const phrases = getTranslatedPhrases(locale);
+
+  const { Identity } = useIdentity();
+
+  useEffect(() => {
+    const getConfig = async () => {
+      await Identity.getConfig()
+        .then((response) => {
+          const {
+            pwLowercase,
+            pwMinLength,
+            pwPwNumbers,
+            pwSpecialCharacters,
+            pwUppercase,
+          } = response;
+
+          setPasswordRequirements({
+            pwLowercase,
+            pwMinLength,
+            pwPwNumbers,
+            pwSpecialCharacters,
+            pwUppercase,
+            status: 'success',
+          });
+        })
+        .catch(() => setPasswordRequirements({ status: 'error' }));
+    };
+
+    if (Identity) {
+      getConfig();
+    }
+  }, [Identity]);
+
+  const {
+    pwLowercase = 0,
+    pwMinLength = 0,
+    pwPwNumbers = 0,
+    pwSpecialCharacters = 0,
+    pwUppercase = 0,
+    status,
+  } = passwordRequirements;
+
+  const passwordErrorMessage = passwordValidationMessage({
+    defaultMessage: phrases.t('identity-block.password-requirements'),
+    options: {
+      lowercase: {
+        value: pwLowercase,
+        message: phrases.t('identity-block.password-requirements-lowercase', { requirementCount: pwLowercase }),
+      },
+      minLength: {
+        value: pwMinLength,
+        message: phrases.t('identity-block.password-requirements-characters', { requirementCount: pwMinLength }),
+      },
+      uppercase: {
+        value: pwUppercase,
+        message: phrases.t('identity-block.password-requirements-uppercase', { requirementCount: pwUppercase }),
+      },
+      numbers: {
+        value: pwPwNumbers,
+        message: phrases.t('identity-block.password-requirements-numbers', { requirementCount: pwPwNumbers }),
+      },
+      specialCharacters: {
+        value: pwSpecialCharacters,
+        message: phrases.t('identity-block.password-requirements-uppercase', { requirementCount: pwUppercase }),
+      },
+    },
+  });
+
+  const handlePasswordUpdate = ({ 'current-password': oldPassword, password: newPassword }) => {
+    if (hasPassword) {
+      return Identity
+        .updatePassword(oldPassword, newPassword)
+        .then(() => {
+          setError(false);
+        }).catch(() => {
+          setError(phrases.t('identity-block.update-password-error'));
+          throw new Error();
+        });
+    }
+    return Identity.signUp(
+      { userName: email, credentials: newPassword },
```

Comment:
you are using email here, didn't see that

Reply:
if you sign in with google, then use a different email without updating the page. then try to create a password without updating the page. the email would need to be the updated email.

the create password flow from a google sign in with a different email would require a verified email. with an unverified one, this process with google sign in becomes unmanageable.